util/package_script_extension.livecodescript

```diff
@@ -0,0 +1,79 @@
+#!
+ 
+send "doPackage" to me
+ 
+on doPackage
+   local tArgs
+   put the commandArguments into tArgs
+
+   if tArgs[11] is true then
+       protectStack the commandName, tArgs[2], tArgs[7], tArgs[3], tArgs[4]
+   end if
+   extractDocs the commandName, tArgs[5], tArgs[6], tArgs[8], tArgs[7]
+   packageScriptExtension tArgs[8], tArgs[9], tArgs[10], tArgs[11]
+
+   quit 0
+end doPackage
+ 
+function q pParam
+    return quote & pParam & quote
+end q
+ 
+command protectStack pEnginePath, pProtectifyScript, pSource, pOutput, pGitHash
+    local tShell
+    put merge(q("[[pEnginePath]]") && q("[[pProtectifyScript]]") && \
+        q("[[pSource]]") && q("[[pOutput]]") && q("[[pGitHash]]")) into tShell
+        
+    get shell(tShell)
+   if the result is not 0 then
+       write "subprocess error:" && it to stderr
+       quit 1
+   end if
+end protectStack
+
+command extractDocs pEnginePath, pExtractScript, pDocsParser, pSourceDir, pSource
+    local tShell
+    put merge(q("[[pEnginePath]]") && q("[[pExtractScript]]") && \
+        q("[[pDocsParser]]") && q("[[pSourceDir]]") && q("[[pSource]]")) \
+        into tShell
+
+    get shell(tShell)
+   if the result is not 0 then
+       write "subprocess error:" && it to stderr
+       quit 2
+   end if
+end extractDocs
+ 
+command packageScriptExtension pSourceFolder, pSourceFileName, pTargetFolder, pRemoveSource   
+
+   local tTargetDir, tBuildTarget, tZip
+   put "com.livecode.script-library." & pSourceFileName into tTargetDir
```

Comment:
can you use `the short name of stack` here?

Reply:
yes!

Reply:
Hmm, actually probably not yet, as the gyp has the dir hard-coded for output